src/main/scala/uk/gov/hmrc/hmrcfrontend/views/config/HmrcPageHeading.scala

```diff
@@ -67,12 +86,12 @@ object HmrcPageHeadingLegend extends HmrcPageHeading {
               """<h1 class="govuk-fieldset__heading hmrc-page-heading govuk-!-margin-top-0 govuk-!-margin-bottom-0">"""
             ),
             content.asHtml,
-            pageHeadingCaption(caption),
+            pageHeadingCaption(caption, makeCaptionSizeClass(classes, HmrcLegendSize)),
```

Comment:
@kyle-bowden what do you think about renaming stuff a bit:
```
govukCaptionClassScaledToMatchHeadingSize(
  headingClasses=classes,
  regexToExtractSizeFromHeadingClasses=GovukFieldsetLegendSize
)
```

Reply:
I like this, I was unsure of how to name it and how verbose I should get.